R/risk_mod.R

```diff
@@ -310,50 +313,74 @@ risk_mod <- function(X, y, gamma = NULL, beta = NULL, weights = NULL,
   if (length(beta) != ncol(X)) stop("beta and X non-compatible")
   if (length(y) != nrow(X)) stop("y and X non-compatible")
 
-  # Run coordinate descent from initial solution
-  res <- risk_coord_desc(X, y, gamma, beta, weights, lambda0, a, b, max_iters,
-                         tol, shuffle)
-  gamma <- res$gamma
-  beta <- res$beta
-
-  # Convert to GLM object
-  glm_mod <- stats::glm(y~.-1, family = "binomial", weights = weights,
-                 start = gamma*beta, method=glm_fit_risk, data = data.frame(X, y))
-  names(beta) <- names(stats::coef(glm_mod))
-
-  # Save model score card
-  nonzero_beta <- beta[beta != 0][-1]
-  if (length(nonzero_beta) <= 1) {
-    model_card <- NULL
-    score_map <- NULL
-  } else {
-    model_card <- data.frame(Points = nonzero_beta)
-
-    # Get range of possible scores
-    X_nonzero <- X[,which(beta != 0)]
-    X_nonzero <- X_nonzero[,-1]
-    min_pts <- rep(NA, length(nonzero_beta))
-    max_pts <- rep(NA, length(nonzero_beta))
-    for (i in 1:ncol(X_nonzero)) {
-      temp <- nonzero_beta[i] * c(min(X_nonzero[,i]), max(X_nonzero[,i]))
-      min_pts[i] <- min(temp)
-      max_pts[i] <- max(temp)
+  if (n_train_runs != round(n_train_runs) | n_train_runs < 0) {
```

Comment:
**question**: Is there a better way to check if an R numeric object is an integer? I was initially using `is.integer(n_train_runs)` but that method [seems faulty](https://stackoverflow.com/questions/3476782/check-if-the-number-is-integer)...